packages/fetchai/contracts/erc1155/contract.py

```diff
@@ -18,20 +18,464 @@
+        }
+
+        # estimate the gas and update the transaction dict
+        gas_estimate = ledger_api.api.eth.estimateGas(transaction=tx)
+        logger.info("gas estimate deploy: {}".format(gas_estimate))
+        tx["gas"] = gas_estimate
+        return tx
+
+    def update_contract_instance(self, contract_address: Address, ledger_api: LedgerApi):
+        """Update the local instance of the smart contract with the deployed one."""
+        logger.info("Updating the local instance of the contract...")
+        self.address = contract_address
+        self.instance = ledger_api.api.eth.contract(address=self.address, abi=self.abi)
+
+    def create_batch(self, deployer_address: Address, ledger_api: LedgerApi) -> TransactionMessage:
+        """
+        Create an mint a batch of items.
+
+        :params address: The address that will receive the items
+        :params mint_quantities: A list[10] of ints. The index represents the id in the item_ids list.
+        """
+        # create the items
+
+        tx = self._get_create_batch_tx(deployer_address=deployer_address, ledger_api=ledger_api)
+
+        #  Create the transaction message for the Decision maker
+        tx_message = TransactionMessage(
+            performative=TransactionMessage.Performative.PROPOSE_FOR_SIGNING,
+            skill_callback_ids=[ContractId("author", "my_skill", "0.1.0")],
+            tx_id="contract_deployment",
+            tx_sender_addr=deployer_address,
+            tx_counterparty_addr="",
+            tx_amount_by_currency_id={"ETH": 0},
+            tx_sender_fee=0,
+            tx_counterparty_fee=0,
+            tx_quantities_by_good_id={},
+            info={},
+            ledger_id="ethereum",
+            signing_payload=tx
+        )
+
+        return tx_message
+
+        # Create a decision_maker transaction message for the contract and return it.
+
+        # tx_signed = self._sign_transaction(tx=tx)
+        # self.send_tx(tx_signed=tx_signed, tx_type="create_batch")
+
+    def _get_create_batch_tx(self, deployer_address: Address, ledger_api: LedgerApi) -> str:
+        """Create a batch of items."""
+        # create the items
+        nonce = ledger_api.api.eth.getTransactionCount(deployer_address)
+        tx = self.instance.functions.createBatch(
+            deployer_address, self.item_ids
+        ).buildTransaction(
+            {
+                "chainId": 3,
+                "gas": 300000,
+                "gasPrice": ledger_api.api.toWei("50", "gwei"),
+                "nonce": nonce,
+            }
+        )
+        logger.info(tx)
+        logger.info(ledger_api.api.provider)
+        # gas_estimate = self.w3.eth.estimateGas(transaction=tx)
+        # logger.info("gas estimate create_batch: {}".format(gas_estimate))
+        return tx
+
+    def mint_batch(self, deployer_address: Address, recipient_address: Address, mint_quantities: List[int], ledger_api: LedgerApi):
```

Comment:
I split the functionalities to different functions so we have a function for create and an other function for minting.